packages/kcmsf/src/pages/match.tsx

```diff
@@ -33,136 +38,135 @@ export const Match = () => {
     },
     [matchJudge, forceReload]
   );
+  const matchStatus: StatusButtonProps["status"] | undefined = useMemo(() => {
+    if (match) {
+      return getMatchStatus(match);
+    }
+    return undefined;
+  }, [match]);
+
   return (
-    <Flex h="100%" direction="column" gap="md" align="center" justify="center">
-      {matchInfo && (
-        <Paper w="100%" p="xs" withBorder>
-          <Flex direction="row" align="center" justify="center">
-            <Text size="2rem" c="blue" flex={1}>
-              {matchInfo?.teams.left?.teamName}
-            </Text>
-            <Flex direction="column" align="center" justify="center" c="dark">
-              {config.match[matchInfo?.matchType].name}
-              <Text size="2rem">#{match?.matchCode}</Text>
-              {match?.matchType == "main" &&
-                `${match.runResults.length == 0 ? 1 : 2}試合目`}
-            </Flex>
-            <Text size="2rem" c="red" flex={1}>
-              {matchInfo?.teams.right?.teamName}
-            </Text>
-          </Flex>
-        </Paper>
-      )}
-      <Button
-        w="100%"
-        h="auto"
-        pb="sm"
-        variant="filled"
-        color={timerState == "finished" ? "pink" : isRunning ? "teal" : "gray"}
-        onClick={switchTimer}
-      >
-        <Text size="5rem">{parseSeconds(totalSeconds)}</Text>
-      </Button>
-      <Paper w="100%" withBorder>
-        <Flex align="center" justify="center">
-          <Button
-            flex={1}
-            variant="transparent"
-            c="blue"
-            leftSection={<IconRotate />}
-            size="xl"
-            fw="normal"
-            onClick={() => onClickReset("left")}
-          >
-            リセット
-          </Button>
-          <Flex pb="sm" gap="sm">
-            <Text size="4rem" c="blue">
-              {isExhibition || matchInfo?.teams.left
-                ? matchJudge.leftTeam.point.point()
-                : 0}
-            </Text>
-            <Text size="4rem">-</Text>
-            <Text size="4rem" c="red">
-              {isExhibition || matchInfo?.teams.right
-                ? matchJudge.rightTeam.point.point()
-                : 0}
-            </Text>
-          </Flex>
+    <>
+      {matchStatus === "end" ? (
+        <MatchResult match={match!} matchInfo={matchInfo!} />
+      ) : (
```

Comment:
non-null assertionは可能な限り使うのを避けてください。
```suggestion
      {match && matchInfo && matchStatus === "end" ? (
        <MatchResult match={match} matchInfo={matchInfo} />
      ) : (
```